src/test/java/com/hillel/items_exchange/controller/AuthControllerIntegrationTest.java

```diff
@@ -0,0 +1,175 @@
+                .andDo(print())
+                .andExpect(status().isUnprocessableEntity())
+                .andReturn();
+
+        assertThat(result.getResolvedException(), is(instanceOf(UnprocessableEntityException.class)));
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void register_whenUsernameExists_shouldReturnUnprocessableEntityAndThrowUnprocessableEntityException()
+            throws Exception {
+
+        UserRegistrationDto invalidUserRegistrationDto = createUserRegistrationDtoWithExistentUsername();
+        MvcResult result = this.mockMvc.perform(post(REGISTER_URL)
+                .content(asJsonString(invalidUserRegistrationDto))
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isUnprocessableEntity())
+                .andReturn();
+
+        assertThat(result.getResolvedException(), is(instanceOf(UnprocessableEntityException.class)));
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void register_whenDifferentConfirmPassword_shouldReturnBadRequestAndThrowBadRequestException()
+            throws Exception {
+
+        UserRegistrationDto invalidUserRegistrationDto = createUserRegistrationDtoWithDifferentPasswords();
+        MvcResult result = this.mockMvc.perform(post(REGISTER_URL)
+                .content(asJsonString(invalidUserRegistrationDto))
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isBadRequest())
+                .andReturn();
+
+        assertThat(result.getResolvedException(), is(instanceOf(BadRequestException.class)));
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void register_whenPasswordInvalid_shouldReturnReturnNotFound()
+            throws Exception {
+
+        UserRegistrationDto invalidUserRegistrationDto = createUserRegistrationDtoWithInvalidPassword();
+        mockMvc.perform(post(REGISTER_URL)
+                .content(asJsonString(invalidUserRegistrationDto))
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    @Transactional
+    @DataSet("database_init.yml")
+    void register_whenEmailInvalid_shouldReturnNotFound()
+            throws Exception {
+
+        UserRegistrationDto invalidUserRegistrationDto = createUserRegistrationDtoWithInvalidEmail();
+        mockMvc.perform(post(REGISTER_URL)
+                .content(asJsonString(invalidUserRegistrationDto))
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isNotFound());
```

Comment:
Why don't tell the user that something is wrong with the request if it has wrong parameter instead of returning not found?